src/client/automation/playback/press/key-press-simulator.js

```diff
@@ -1,26 +1,31 @@
 import hammerhead from '../../deps/hammerhead';
 import { KEY_MAPS, domUtils, getSanitizedKey } from '../../deps/testcafe-core';
 import typeChar from '../type/type-char';
-import { getChar, getDeepActiveElement } from './utils';
+import { getChar, getDeepActiveElement, changeLetterCase } from './utils';
 import getKeyCode from '../../utils/get-key-code';
+import getKeyIdentifier from '../../utils/get-key-identifier';
+import isLetterKey from '../../utils/is-letter';
 
 var browserUtils   = hammerhead.utils.browser;
 var extend         = hammerhead.utils.extend;
 var eventSimulator = hammerhead.eventSandbox.eventSimulator;
 
-var isSafariOrAndroid = browserUtils.isSafari || browserUtils.isAndroid;
+var useKeyIdentifier = browserUtils.isSafari || browserUtils.isAndroid && browserUtils.version < 54;
 
 export default class KeyPressSimulator {
-    constructor (key) {
+    constructor (key, originKey) {
```

Comment:
For expression 'shift+!', in case with second symbol '!' - `key` will be '1', `originKey` will be '!'. `key` needed to retrieve the button, what we actually press (button '1'), `originKey` needed to fill `event.key` property or `event.keyIdentifier`